### DISCUSSION TITLE
fatal: Bad Object error

### DISCUSSION THREAD

**Discussion by asishrj91** [community/community#65773] [Actions] [answered]
### Select Topic Area

Question

### Body

**Hi,
My objective of using the below github yaml code is to copy the files which get modified in the source repository(test1) to the destination repository(Test2)**

name: Move Modified Files

on:
  push:
    branches:
      - main

jobs:
  move-files:
    runs-on: ubuntu-latest

    steps:
          
      - name: Checkout code
        uses: actions/checkout@v2

      - name: Set Up Git
        run: |
          git config --global user.name "asishrj91"
          git config --global user.email "asishrj91@gmail.com"

      - name: List Modified Files
        id: git-changes
        run: |
          git diff --name-status ${{ github.event.before }} ${{ github.sha }} > changed-files.txt
          cat changed-files.txt

      - name: Clone Destination Repo
        run: |
          git clone https://$USERNAME:$PAT_SOURCE_REPO@github.com/asishrj91/Test2.git destination-repo
        env:
          USERNAME: asishrj91
          PAT_SOURCE_REPO: ${{ secrets.PAT_SOURCE_REPO }}
        

      - name: Move Modified Files to Destination Repo
        run: |
          xargs -I {} sh -c 'mkdir -p destination-repo/$(dirname {}); mv {} destination-repo/{}' < changed-files.txt
          cd destination-repo
          git add .
          git commit -m "Move modified files"
          git push origin main
        working-directory: .
        
**But i'm getting the below error:**

Run git diff --name-status 1168b9cde55971e56ec8f1faec6de79601[2](https://github.com/asishrj91/test1/actions/runs/6059189619/job/16442043343#step:4:2)31752 5e742fbf6517a01dcebbad8dab27153748cb1f2a > changed-files.txt
  git diff --name-status 1168b9cde55971e56ec8f1faec6de79601231752 5e742fbf6517a01dcebbad8dab27153748cb1f2a > changed-files.txt
  cat changed-files.txt
  shell: /usr/bin/bash -e {0}
fatal: bad object 1168b9cde55971e56ec8f1faec6de796012[3](https://github.com/asishrj91/test1/actions/runs/6059189619/job/16442043343#step:4:3)17[5](https://github.com/asishrj91/test1/actions/runs/6059189619/job/16442043343#step:4:6)2
Error: Process completed with exit code 128.
![image](https://github.com/community/community/assets/53378614/6d4f59cc-6efc-4d1e-8615-c36950638fb9)
**I could see all the commits in Github as well**
![image](https://github.com/community/community/assets/53378614/cb9ea67c-1d20-4a80-b233-e1a95063250b)

**Why I am getting this ? Can anyone help me ?**

**Comment by arafajs** [comment, marked as answer]
- Use a different reference for the before commit, like the previous branch head rather than a specific SHA:
`git diff --name-status @{u} ${{ github.sha }} > changed-files.txt`

- Fetch git history deeper than just the last commit to make sure the before commit is available:
```

- uses: actions/checkout@v2
+ uses: actions/checkout@v2
+     with:
+       fetch-depth: 0
```

- If the before commit shouldn't change, you could hardcode a known valid SHA instead of using the variable ${{ github.event.before }} This should resolve the error and allow git diff to compare the correct commits.

**Reply by asishrj91** [reply]
uses: actions/checkout@v2
+     with:
+       fetch-depth: 0

This worked. Thanks @arafajs
